sympy/physics/units/quantities.py

```diff
@@ -88,6 +88,10 @@ def _eval_is_positive(self):
     def _eval_is_constant(self):
         return self.scale_factor.is_constant()
 
+    def _eval_Abs(self):
+        return self.__class__(self.name, self.dimension,
```

Comment:
We have `self.func` in SymPy for this cases. `self.func` is to prefer to `type(self)` and `self.__class__` because there are some proposals to remove the metaclasses currently used.